### DISCUSSION TITLE
Pagination UI issue

### DISCUSSION THREAD

**Discussion by kabyanil** [livewire/livewire#3365] [1. Help]
Hi,

I'm using livewire pagination to display a list of around 500 profiles. The UI breaks like this -

<img width="815" alt="Screenshot 2021-07-19 at 12 15 14 AM" src="https://user-images.githubusercontent.com/24878109/126078785-1b7d7088-6b53-4809-b921-d5366b18fabc.png">

Can anyone help me fix this?

Thanks!

**Comment by ousid** [comment]
Hey, @kabyanil 

First Publish the assets 
```
php artisan livewire:publish --assets
```

And then go to `resources/views/vendor/liveiwre/pagination` then you can fix this issue or even redesign your pagination however you want.

Let me know if your problem has been solved.

**Reply by kabyanil** [reply]
thanks for the quick reply!

i ran the command, but it didn't generate the livewire folder inside resources/views/vendor/ folder.

**Comment by mikethetreeclimber** [comment]
Have you tried just using *simplePaginate()* ?  

**Try**
```php artisan livewire:publish --pagination```

Or you can call a custom view directly in the blade file
```{{ $posts->links('custom-pagination-links-view') }}```

The docs also give a method to override the default pagination view directly in the component class.

**Reply by kabyanil** [reply]
Thanks for the reply!

I tried your command. It generated the pagination UI blade file, which I can customize.

It would be nice to have an option to limit the number of pages to show in the pagination bar. For instance, if I could specify a limit of ten buttons to display, that would be nice. Please let me know if something exists already.

**Reply by mikethetreeclimber** [reply]
I would start by wrapping the pagination in a section tag or some html tag and playing with the width of the element to make it a more suitable.  You could also @dump($somevariable->links()) and see what data is available  to you and create logic around that

**Reply by mikethetreeclimber** [reply]
If you look at the docs, livewire provides you with the ability to override the method used for pagination.  
https://laravel-livewire.com/docs/2.x/pagination

**Comment by ph7jack** [comment]
You can use **onEachSide** function to limit the page numbers rendered on pagination links
https://laravel.com/docs/8.x/pagination#adjusting-the-pagination-link-window

**Reply by mikethetreeclimber** [reply]
This is why I love php and especially love laravel, so many of the titious things that are basically standard in UIs are already beautifully implemented allowing us to focus on the unique logic specific to our applications.  And Now we are given livewire to wrap around laravel. 💦
